sgt-back-end/index.js

```diff
@@ -0,0 +1,154 @@
+const pg = require('pg');
+const express = require('express');
+var app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades/', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    }).catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+
+});
+
+app.post('/api/grades/', (req, res) => {
+  const newGradeObj = req.body;
+  if (!newGradeObj.name || !newGradeObj.course || !newGradeObj.score) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
```

Comment:
A `score` of `0` should not trigger a `400 Bad Request`.

Reply:
<img width="1270" alt="Screen Shot 2022-08-26 at 11 17 27 AM" src="https://user-images.githubusercontent.com/106217355/186967236-fdf905b6-d8f5-4673-9035-24ea42daa664.png">

Reply:
<img width="927" alt="Screen Shot 2022-08-26 at 11 18 37 AM" src="https://user-images.githubusercontent.com/106217355/186967372-bfd13662-792b-494a-bc1f-d64786bae89d.png">